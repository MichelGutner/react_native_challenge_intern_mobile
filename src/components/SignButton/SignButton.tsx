import React from 'react';
import styled from 'styled-components/native';
import { Themes } from '../../../themes/themes';

const Button = styled.TouchableOpacity`
  width: 312px;
  height: 48px;
  left: 24px;
  top: 96px;
  border-radius: 4px;
  background-color: ${Themes.colors.white};
  align-items: center;
  justify-content: center;
`;

const TitleButton = styled.Text`
  font-family: Roboto;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;

  display: flex;
  align-items: center;
  text-align: center;

  color: ${Themes.colors.backgroundSanar};
`;

type TextProps = {
  title: string;
  onPress: any;
};

const SignButton = ({ title, onPress, ...rest }: TextProps) => {
  return (
    <Button {...rest} onPress={onPress}>
      <TitleButton>{title}</TitleButton>
    </Button>
  );
};

export default SignButton;
