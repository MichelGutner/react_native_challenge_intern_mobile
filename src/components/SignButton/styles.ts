import styled from 'styled-components/native';
import { Themes } from '../../../themes/themes';

export const Button = styled.TouchableOpacity`
  width: 312px;
  height: 48px;
  left: 24px;
  top: 96px;
  border-radius: 4px;
  background-color: ${Themes.colors.white};
  align-items: center;
  justify-content: center;
`;

export const TitleButton = styled.Text`
  font-family: Roboto;
  font-style: normal;
  font-weight: 500;
  font-size: 16px;

  display: flex;
  align-items: center;
  text-align: center;

  color: ${Themes.colors.backgroundSanar};
`;
