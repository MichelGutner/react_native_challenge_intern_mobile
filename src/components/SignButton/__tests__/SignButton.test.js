import React from 'react';
import renderer from 'react-test-renderer';
import SignButton from '../SignButton';

test('renders correctly', () => {
  const tree = renderer.create(<SignButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
