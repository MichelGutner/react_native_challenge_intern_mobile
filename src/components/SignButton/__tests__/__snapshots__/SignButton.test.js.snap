// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFF",
      "borderRadius": 4,
      "height": 48,
      "justifyContent": "center",
      "left": 24,
      "opacity": 1,
      "top": 96,
      "width": 312,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "alignItems": "center",
          "color": "#0050F0",
          "display": "flex",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "fontStyle": "normal",
          "fontWeight": "500",
          "textAlign": "center",
        },
      ]
    }
  />
</View>
`;
